### ISSUE TITLE
Video description is truncated

### ISSUE THREAD

**Issue by zimawho** [trizen/straw-viewer#6]
The descriptions are truncated in different ways. 

Sometimes, only urls are truncated. For example, in [this video](https://www.youtube.com/watch?v=sNhhvQGsMEc).
```
Also, for more in depth information take a look at the WAIT BUT WHY article about Fermi Paradox:
http://waitbutwhy.com/2014/05/fermi-p...
```
In [another video](https://www.youtube.com/watch?v=H6u0VBqNBQ8), the whole second part of the description is missing
```
Sources and link to book by Rupert Glasgow:
https://sites.google.com/view/sources-consciousness

Consciousness is perhaps the ...
```
I checked on invidio.us and the descriptions of these videos seem complete. Any idea how to fix that in straw-viewer ?

**Comment by trizen** [comment]
The description displayed by straw-viewer is the description retrieved from the API of invidious.

In some contexts, the description is indeed truncated. For example, in `/search` results, the description is truncated more:

https://invidio.us/api/v1/search?maxResults=20&part=snippet&q=The%20Origin%20of%20Consciousness&type=video&pretty=1

compared to the description retrieved in `/videos` context:
https://invidio.us/api/v1/videos/H6u0VBqNBQ8?fields=description&pretty=1

But, some long URLs are truncated even in the extended video info:
https://invidio.us/api/v1/videos/sNhhvQGsMEc?fields=description&pretty=1

However, invidious provides the `descriptionHtml` field, which includes the full URLs:
https://invidio.us/api/v1/videos/sNhhvQGsMEc?fields=descriptionHtml&pretty=1

Maybe `straw-viewer` should try to extract the URLs from `descriptionHtml`?

Additionally, it may be possible to extract the full description for each video in search results, when the video is selected for play, but this would slow things down a little bit, as each request to `/video` seems to take several seconds.

Note that `straw-viewer` already does this, but only when given a video URL:

```console
$ straw-viewer 'https://www.youtube.com/watch?v=H6u0VBqNBQ8'
```

**Comment by zimawho** [comment]
Thanks for the explanation over invidious' API. I understand the root of the problem now. I used youtube-viewer and now straw-viewer in a macro in my CLI-rss client. Having truncated urls seems to defeat the purpose of this use case because I still need to open a webpage to retrieve the complete urls.
Extracting the urls from `descriptionHtml` seems like a good idea.

**Comment by zimawho** [comment]
It seems to be fixed with v0.0.3, thanks !
